### PR TITLE
Added noAuth page and routing

### LINKED ISSUE #1394: hub: display something if the user does not have read permissions
The Cylc UI Server authorisation system blocks restricts API calls to authorised users, static file requests, however, are not subject to authorisation.

This means that if a user does not have `READ` permissions, the Cylc UI app will load, but no workflows will show up. This makes it look like the Cylc UI is broken.

> Note: It's the same story with Jupyter Lab, which will also load the app, but all API calls will fail.

We should try to make this situation a little more obvious to the authenticated user by displaying a static page saying something like "you are not authorised to access this server" in the Cylc UI.

### PR DESCRIPTION
**Pull request by markgrahamdawson**
Closes #1394

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

File: tests/e2e/specs/url.cy.js [hunk 1]
```diff
@@ -35,4 +35,21 @@ describe('URL handling', () => {
         expect(url.endsWith('?a=1&b=2')).to.be.true
       })
   })
+
+  it('reroutes to noAuth page if user isnt authenticated', () => {
```
Comment (MetRonnie) [inline tests/e2e/specs/url.cy.js:39]: Difference between authorisation and authentication is basically
- Authentication is ensuring the identity of the user
- Authorisation is determining the permissions of the user

In this case the user is authenticated but not authorised to view anything
```suggestion
  it('reroutes to noAuth page if user isnt authorised', () => {
```

**Review by oliver-sanders — APPROVED** [review]

**Comment by MetRonnie** [comment]
For some reason this `analysis.cy.js` test seems to be reliably failing on this branch but only for Firefox

Analysis view -- sorts the chart entries -- before each hook (failed)

File: src/router/paths.js [hunk 2]
```diff
@@ -145,5 +145,15 @@ export default [
...
     props: true
+  },
+  {
+    path: '/noAuth',
+    view: 'NoAuth',
+    name: 'noAuth',
+    meta: {
+      layout: 'noAuth',
+      toolbar: false,
+      showSidebar: true
```
Comment (MetRonnie) [inline src/router/paths.js:156]: (Don't need these for this custom layout)
```suggestion
```

**Comment by oliver-sanders** [comment]
> For some reason this analysis.cy.js test seems to be reliably failing on this branch but only for Firefox

It passed on retry, this test is flaky, I've put up an issue, can ignore for now.

**Pull request merged**